### PR TITLE
db-3566 [all FE starters] Log helpful message if no env vars are set

### PR DESCRIPTION
**Pull request by lotusbaba**
## What changes were made?
Changes to provide meaningful error messages when critical environment variables aren't present

## Where were the changes made?
 
 
Changes to files - `starters/gatsby-wordpress-starter/gatsby-config.js`, `starters/next-wordpress-starter/next.config.js`, & `starters/next-drupal-starter/next.config.js`

## How have the changes been tested?
Yes on http://localhost

## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

**Review by CobyPear — CHANGES_REQUESTED** [review]
See comments. Also, I don't think the `try/catch` in necessary. If you throw in error in the config it will exit the process appropriately.

File: starters/next-drupal-starter/next.config.js [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
-}
-// remove trailing slash if it exists
-imageDomain = imageDomain.replace(/\/$/, "");
+  require("dotenv").config({
+    path: path.resolve(process.cwd(), ".env.development.local"),
+  });
+
+  if (process.env.IMAGE_DOMAIN === undefined) {
+    throw new Error(
+      "Please set IMAGE_DOMAIN in the .env.development.local file"
```
[the remark this answers is not in the document]
Comment (lotusbaba) [reply starters/next-drupal-starter/next.config.js]: Done

File: starters/gatsby-wordpress-starter/gatsby-config.js [hunk 2]
```diff
@@ -1,14 +1,24 @@
 const path = require("path")
-require("dotenv").config({
-  path: path.resolve(process.cwd(), ".env.local"),
-})
+const fs = require("fs")
+try {
+  if (!fs.existsSync(path.resolve(process.cwd(), ".env.development.local"))) {
```
[the remark this answers is not in the document]
Comment (lotusbaba) [reply starters/gatsby-wordpress-starter/gatsby-config.js]: Done

File: starters/gatsby-wordpress-starter/gatsby-config.js [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
 // Pantheon CMS endpoint
+
+if (
+  process.env.WPGRAPHQL_URL === undefined &&
+  process.env.PANTHEON_CMS_ENDPOINT === undefined
+) {
+  throw new Error(
+    "Please set critical environment variables PANTHEON_CMS_ENDPOINT or WPGRAPHQL_URL."
+  )
+}
```
Comment (CobyPear) [inline starters/gatsby-wordpress-starter/gatsby-config.js:18]: The check for the env var looks good, but the error message still doesn't pass the AC on the ticket:

- the error message should suggest ways to fix the problem by checking the NODE_ENV and suggesting an appropriate fix for each 

  - set a .env.development.local with the critical envs for NODE_ENV=development

  - link a backend or set the critical env var in the dashboard for NODE_ENV=production

**Review by CobyPear — CHANGES_REQUESTED** [review]

**Pull request closed without merging**